Tests/SwiftLintFrameworkTests/ConfigurationTests.swift

```diff
@@ -195,6 +195,22 @@ class ConfigurationTests: XCTestCase {
                           projectMockConfig0CustomPath.configuration(for: file))
     }
 
+    func testConfigurationWithSwiftFileAsRoot() {
+        let configuration = Configuration(path: projectMockYAML0,
+                                          rootPath: projectMockSwift0,
+                                          optional: false, quiet: true)
+        let file = File(path: projectMockSwift0)!
+        XCTAssertEqual(configuration.configuration(for: file), configuration)
+    }
+
+    func testConfigurationWithSwiftFileAsRootAndCustomConfiguration() {
+        let configuration = Configuration(path: projectMockYAML0CustomPath,
+                                          rootPath: projectMockSwift0,
+                                          optional: false, quiet: true)
+        let file = File(path: projectMockSwift0)!
+        XCTAssertEqual(configuration.configuration(for: file), configuration)
```

Comment:
> @marcelofabri,

For starters, `Configuration`’s `Equatable` conformance is broken on master, so `XCTAssertEqual` is unreliable here. (See the comments in #1748 starting with [this one](https://github.com/realm/SwiftLint/pull/1748#issuecomment-320370564)). Although that may not be the only issue since there are also failures on macOS.